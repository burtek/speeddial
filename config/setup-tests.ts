import '@testing-library/jest-dom';
import { createSerializer } from '@emotion/jest';
import i18n from 'i18next';
import 'jest-extended';
import 'jest-extended/all';
import { initReactI18next } from 'react-i18next';


expect.addSnapshotSerializer(createSerializer());

await i18n
    .use(initReactI18next)
    .init({
        lng: 'dev',
        interpolation: { escapeValue: false }
    });
