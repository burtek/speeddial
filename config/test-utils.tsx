import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { FC, PropsWithChildren, ReactElement } from 'react';
import { Provider as StoreProvider } from 'react-redux';

import { store } from '@@data/index';
import { ThemedApp } from 'src/theme/themed-app';

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => (
    <StoreProvider store={store}>
        <ThemedApp>
            {children}
        </ThemedApp>
    </StoreProvider>
);

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';

// override render method
// eslint-disable-next-line import/export
export { customRender as render };
