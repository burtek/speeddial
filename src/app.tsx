import { Typography } from '@mui/material';
import Box from '@mui/material/Box';

import { SearchBar } from './components/search';

export function AppLayout() {
    // eslint-disable-next-line no-warning-comments
    // TODO: flex layout, not grid layout
    return (
        <Box
            sx={{
                display: 'grid',
                gridTemplateAreas: `
                    "left-sidebar   search   right-sidebar"
                    "left-sidebar   main     right-sidebar"
                    "spacer         spacer   spacer"
                    "footer         footer   footer"`,
                gridTemplateColumns: '1fr 1150px 1fr',
                gridTemplateRows: 'min-content min-content 1fr min-content',
                gap: '2vw',
                minHeight: 'calc(100vh - 20px)'
            }}
        >
            <SearchBar gridArea="search" />
            <div style={{ gridArea: 'main' }} />
            <div style={{ gridArea: 'spacer' }} />
            <Box sx={{ /* position: 'absolute', bottom: 0, left: 0, right: 0 */ gridArea: 'footer', textAlign: 'center' }}>
                <Typography>
                    &copy;
                    {' '}
                    <a
                        href="https://github.com/burtek"
                        target="_blank"
                        referrerPolicy="no-referrer"
                        rel="noopener noreferrer nofollow"
                    >
                        burtek
                    </a>
                    {` ${new Date().getFullYear()} | `}
                    <a
                        href="https://github.com/burtek/speeddial"
                        target="_blank"
                        referrerPolicy="no-referrer"
                        rel="noopener noreferrer nofollow"
                    >
                        This project is open-source!
                    </a>
                    {` | Build ${import.meta.env.VITE_VERCEL_GIT_COMMIT_SHA}`}
                </Typography>
            </Box>
        </Box>
    );
}
