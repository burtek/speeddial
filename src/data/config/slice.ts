import type { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from '@reduxjs/toolkit';

export type ThemeMode = 'dark' | 'light';

export const initialState = () => ({
    modalOpen: false,
    themeMode: null as ThemeMode | null
});

export const { actions, name: sliceName, reducer } = createSlice({
    name: 'config',
    initialState,
    reducers: {
        openModal(state) {
            state.modalOpen = true;
        },
        closeModal(state) {
            state.modalOpen = false;
        },

        setThemeMode(state, { payload }: PayloadAction<{ mode: ThemeMode | null }>) {
            state.themeMode = payload.mode;
        }
    }
});
