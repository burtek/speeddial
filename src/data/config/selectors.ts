import { createSelector } from '@reduxjs/toolkit';

import type { RootState } from '..';

import { sliceName } from './slice';


export const getConfig = (state: RootState) => state[sliceName];

export const getThemeMode = createSelector(getConfig, s => s.themeMode);

export const getIsDialogOpen = createSelector(getConfig, s => s.modalOpen);
