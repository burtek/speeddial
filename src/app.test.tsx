import { render } from 'config/test-utils';

import { AppLayout } from './app';


describe('app', () => {
    it('should match snapshot', () => {
        const { container } = render(<AppLayout />);

        expect(container).toMatchSnapshot();
    });
});
