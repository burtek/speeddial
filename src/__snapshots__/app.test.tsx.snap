// Vitest Snapshot v1

exports[`app > should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1a18wfb"
  >
    <div
      class="MuiBox-root css-d8kh1e"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CogIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-isl53j-MuiPaper-root"
    >
      <form
        action="https://www.google.com/search"
        method="get"
        style="display: flex;"
      >
        <input
          name="q"
          placeholder="search.placeholder"
          style="flex: 1; padding: 0.3vw 0.5vw; border-radius: 5px;"
          type="text"
        />
      </form>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-bcshru-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-a66f6b"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1tcztm3-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-k3u79o-MuiCardContent-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              color="grey"
              data-testid="AddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      id="DndDescribedBy-0"
      style="display: none;"
    >
      
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="DndLiveRegion-0"
      role="status"
      style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
    />
    <div
      style="grid-area: spacer;"
    />
    <div
      class="MuiBox-root css-1xm93o3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Â©
         
        <a
          href="https://github.com/burtek"
          referrerpolicy="no-referrer"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          burtek
        </a>
         2023 | 
        <a
          href="https://github.com/burtek/speeddial"
          referrerpolicy="no-referrer"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          footer.openSource
        </a>
         | Build test
      </p>
    </div>
  </div>
</div>
`;
